pulsar-impl/src/main/java/com/datastax/oss/sink/pulsar/AvroTypeUtil.java

```diff
@@ -15,24 +15,74 @@
  */
 package com.datastax.oss.sink.pulsar;
 
+import java.nio.ByteBuffer;
+import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import org.apache.avro.Conversion;
+import org.apache.avro.Schema;
 import org.apache.pulsar.client.api.schema.GenericRecord;
 import org.apache.pulsar.common.schema.SchemaType;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 public final class AvroTypeUtil {
 
-  private static final Logger log = LoggerFactory.getLogger(AvroTypeUtil.class);
+  private static Map<String, Conversion<?>> logicalTypeConverters = new HashMap<>();
+
+  static {
+    logicalTypeConverters.put(
+        CqlLogicalTypes.CQL_DECIMAL, new CqlLogicalTypes.CqlDecimalConversion());
+    logicalTypeConverters.put(
+        CqlLogicalTypes.CQL_DURATION, new CqlLogicalTypes.CqlDurationLogicalType());
+    logicalTypeConverters.put(
+        CqlLogicalTypes.CQL_VARINT, new CqlLogicalTypes.CqlVarintConversion());
+  }
 
   private AvroTypeUtil() {}
 
   public static boolean shouldWrapAvroType(GenericRecord record, Object fieldValue) {
     return record != null && record.getSchemaType() == SchemaType.AVRO && isMapOrList(fieldValue);
   }
 
+  public static boolean shouldHandleLogicalType(GenericRecord record, Object fieldValue) {
+    if (record == null && record.getSchemaType() != SchemaType.AVRO) {
+      return false;
+    }
+    if (isVarint(fieldValue)) {
+      return true;
+    } else if (fieldValue instanceof GenericRecord
+        && ((GenericRecord) fieldValue).getNativeObject()
+            instanceof org.apache.avro.generic.GenericRecord) {
+      org.apache.avro.generic.GenericRecord avroRecord =
+          (org.apache.avro.generic.GenericRecord) ((GenericRecord) fieldValue).getNativeObject();
+      return logicalTypeConverters.containsKey(avroRecord.getSchema().getName());
+    }
+    return false;
+  }
+
+  public static Object handleLogicalType(Object value) {
```

Comment:
I would call this "handleCassandraCDCLogicalType", this way it is a hard reference to CDC
adding a link to the docs of CDC would help as well probably